### PR TITLE
Move otlpmetrics `Client` to an internal package.

### LINKED ISSUE #3455: Do NOT expose generated protos in metrics exporter
See https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlpmetric/client.go and just remember your issue from 6 months (changes can happen in the generated code that are not backwards compatible from golang perspective).

### PR DESCRIPTION
**Pull request by MadVikingGod**
Resolves #3455

This was discussed in the SIG meeting. We want to move the `exporters/otlp/otlpmetric.Client` interface into an internal package.  This allows for the current common logic to be shared between otlpmetrichttp and otlpmetricgrpc, but allows us to expose this in the future if needed. 

Users of `otlpmetric.New()` will need to use the `otlpmetric[http|grpc].New()` directly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Do not report empty partial-success responses in the `go.opentelemetry.io/otel/exporters/otlp` exporters. (#3438, #3432)
 - Handle partial success responses in `go.opentelemetry.io/otel/exporters/otlp/otlpmetric` exporters. (#3162, #3440)
 
+## Removed
+
+- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.Client` interface is removed. (#????)
+- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.New` function is removed. Use the `otlpmetric[http|grpc].New` directly. (#????)
```
Comment (MrAlias) [inline CHANGELOG.md:59]:
```suggestion
- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.Client` interface is removed. (#3486)
- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.New` function is removed. Use the `otlpmetric[http|grpc].New` directly. (#3486)
```

File: exporters/otlp/otlpmetric/internal/exporter.go [hunk 2]
```diff
@@ -12,8 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package otlpmetric // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
-
+package internal // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal"
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/internal/exporter.go:15]:
```suggestion
package internal // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal"

```

**Review by MrAlias — APPROVED** [review]
LGTM when minor issues resolved.

**Review by jmacd — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -53,11 +53,17 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+
+- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.Client` interface is removed. (#3486)
+- The `go.opentelemetry.io/otel/exporters/otlp/otlpmetric.New` function is removed. Use the `otlpmetric[http|grpc].New` directly. (#3486)
+
 ### Deprecated
 
 - The `go.opentelemetry.io/otel/sdk/metric/view` package is deprecated.
   Use `Instrument`, `InstrumentKind`, `View`, and `NewView` in `go.opentelemetry.io/otel/sdk/metric` instead. (#3476)
 
+
```
Comment (MrAlias) [inline CHANGELOG.md:66]:
```suggestion
```

**Pull request merged**